packages/core/src/shared/sshConfig.ts

```diff
@@ -178,6 +178,7 @@ Host ${this.configHostName}
     AddKeysToAgent yes
     StrictHostKeyChecking accept-new
     ProxyCommand ${proxyCommand}
+    IdentitiesOnly yes
```

Comment:
The insight + URL mentioned in the solution, would be useful agumentation of the comment on line 172. Also worth putting in the commit message.